pages/bridge/index.tsx

```diff
@@ -1,219 +1,36 @@
-          setModalState("wrongNetwork");
-          setModalOpen(true);
-          break;
-      }
+  useEffect(() => {
+    const ethereumNetwork = window.localStorage.getItem("ethereum-network");
+    ethereumNetwork ? setBridgeState("emery") : setBridgeState("connect");
+  }, []);
 
-      window.localStorage.setItem("ethereum-network", ethereumNetwork);
-      updateNetwork(ethereum, ethereumNetwork);
-    } catch (err) {
-      setModalState("noMetamask");
-      setModalOpen(true);
-    }
-  };
-
-  const CENNZnetButton = (
-    <Frame
-      sx={{
-        top: "40%",
-        cursor: "pointer",
-        backgroundColor: "#FFFFFF",
-        height: "60px",
-        width: "70%",
-      }}
-      onClick={() => {
-        connectWallet();
-        setModalState("changeAccount");
-        setModalOpen(true);
-      }}
-    >
-      {selectedAccount ? (
-        <>
-          <Heading
-            sx={{
-              color: "primary.main",
-              ml: "10px",
-              mt: "3px",
-              fontSize: "20px",
-              flexGrow: 1,
-              textTransform: "none",
-            }}
-          >
-            CENNZnet
-          </Heading>
-          <SmallText
-            sx={{
-              color: "black",
-              fontSize: "16px",
-              whiteSpace: "nowrap",
-              overflow: "hidden",
-              ml: "5px",
-            }}
-          >
-            {selectedAccount.name}
-          </SmallText>
-        </>
-      ) : (
-        <Heading
-          sx={{
-            fontSize: "20px",
-            margin: "0 auto",
-            color: "primary.main",
-            textTransform: "none",
-          }}
-        >
-          CONNECT CENNZnet WALLET
-        </Heading>
-      )}
-    </Frame>
-  );
-
-  const MetamaskButton = (
-    <Frame
-      sx={{
-        display: "flex",
-        mt: "8%",
-        width: "70%",
-        height: "60px",
-        cursor: "pointer",
-        backgroundColor: "white",
-      }}
-      onClick={connectMetamask}
-    >
-      {Account ? (
-        <>
-          <Heading
-            sx={{
-              color: "primary.main",
-              ml: "10px",
-              mt: "3px",
-              fontSize: "20px",
-              flexGrow: 1,
-            }}
-          >
-            METAMASK
-          </Heading>
-          <SmallText sx={{ color: "black", fontSize: "16px" }}>
-            {Account.substr(0, 6).concat(
-              "...",
-              Account.substr(Account.length - 4, 4)
-            )}
-          </SmallText>
-        </>
-      ) : (
-        <Heading
-          sx={{
-            color: "primary.main",
-            fontSize: "20px",
-            margin: "0 auto",
-          }}
-        >
-          CONNECT METAMASK WALLET
-        </Heading>
-      )}
-    </Frame>
+  return (
+    <CENNZApiProvider>
+      <DappModuleProvider>
+        <Web3AccountsProvider>
+          <BlockchainProvider>
+            <SupportedWalletProvider>
```

Comment:
Should we move these providers on the `_app.tsx` level?

Reply:
done, I left BlockchainProvider in bridge/index as it will only be used there